cmd/kube-scheduler/app/options/options.go

```diff
@@ -208,6 +215,11 @@ func (o *Options) ApplyTo(c *schedulerappconfig.Config) error {
 		}
 	}
 	o.Metrics.Apply()
+
+	if err := utilfeature.DefaultMutableFeatureGate.SetFromMap(c.ComponentConfig.FeatureGates); err != nil {
+		return err
+	}
```

Comment:
because feature gates can be checked by almost anything, I would set them as early as possible (before SecureServing.ApplyTo, etc)